Automattic/wp-calypso
# Partner Portal: Add an agency sign up form

PT: pdpAdu-if-p2
Task: 1202074155672006-as-1202253703889501
Requires D80946-code.

Scope changes in the interest of launching this faster:
- The "Name" and "Email" fields have been removed as we already have that info as the user has to be registered to reach this form.
- The Company URL field has been removed as we don't have a suitable datastore for this at this time.

#### Changes proposed in this Pull Request

* Partner Portal: Add an agency signup form.

#### Testing instructions

* Apply PR and WPCOM patch, sandbox the API.
* Run Jetpack Cloud.
* Open http://jetpack.cloud.localhost:3000/agency/signup
* Interact and test the form. Since only one partner account is allowed per WPCOM account you will most likely hit a "You are already registered as an agency." error. You may want to disable this check in D80946-code in order to create multiple partner accounts while testing but don't go overboard as we don't delete partner accounts at this time.

![Screen Shot 2022-05-27 at 17 10 10](https://user-images.githubusercontent.com/22746396/170716222-f8a267b0-9efe-4c8d-98ef-9d6a58a2a5b2.png)
